### ISSUE TITLE
Upgrading from Tasmota 6.5.0.8 failed with WiFi not connecting

### ISSUE THREAD

**Issue by Crazor** [stefanbode/Sonoff-Tasmota#167]
### BUG DESCRIPTION
I wanted to switch a Sonoff Dual R2 with Tasmota 6.5.0.8 to this fork. I tried to build and upload the normal sonoff/firmware.bin and got an error. Then I remembered that I had to upload sonoff-minimal/firmware.bin on another device before to work around this. So I built and uploaded sonoff-minimal/firmware.bin via the webinterface. Unfortunately, the device did not connect to wifi again and also did not bring up the wifi manager. Pushing the button 4 times (or shorting the other button pin to GND on the header next to the serial port on the Dual R2) did not bring up the WiFi manager either.

I had to then disconnect the device and bring it up on my bench to reflash with esptool.py, which worked fine. I did get a glimpse of the serial output before reflashing, but unfortunately, I managed to copy the logs and not save them anywhere. There were a bunch of messages about the R2 trying to connect to the AP and ultimately failing after several attempts. Both LEDs were on.

Is this a bug or is that behaviour expected when switching from mainline Tasmota to this fork? Is there any way to work around this issue? I've got a couple of Shelly 2.5's that I'd also like to upgrade, but those are a pain to disconnect or even reach to press a button if something goes wrong...

### REQUESTED INFORMATION
- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Sonoff-Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Sonoff-Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in issues (https://github.com/arendst/Sonoff-Tasmota/issues)
- [x] Searched the problem in the wiki (https://github.com/arendst/Sonoff-Tasmota/wiki/Troubleshooting)
- [x] Searched the problem in the forum (https://groups.google.com/d/forum/sonoffusers)
- [x] Searched the problem in the chat (https://discord.gg/Ks2Kzd4)
- [x] Device used (i.e. Sonoff Basic) : Sonoff Dual R2
- [x] Tasmota binary firmware version number used : 6.6.0.1 stb-1.5 (self-compiled)
- [x] Development IDE - Compiler / Upload tools used : platformio / esptool.py
- [x] Provide the output of command ``status 0`` :
```
0:59:48 CMD: status 0
10:59:48 RSL: stat/sonoff/STATUS = {"Status":{"Module":39,"FriendlyName":["Sonoff","Sonoff2"],"Topic":"sonoff","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0}}
10:59:48 RSL: stat/sonoff/STATUS1 = {"StatusPRM":{"Baudrate":115200,"GroupTopic":"sonoffs","OtaUrl":"http://thehackbox.org/tasmota/release/sonoff.bin","RestartReason":"Software/System restart","Uptime":"0T00:14:12","StartupUTC":"2019-07-28T09:45:36","Sleep":50,"CfgHolder":4617,"BootCount":4,"SaveCount":18,"SaveAddress":"FA000"}}
10:59:48 RSL: stat/sonoff/STATUS2 = {"StatusFWR":{"Version":"6.6.0.1 stb-1.5(sonoff)","BuildDateTime":"2019-07-28T10:06:28","Boot":31,"Core":"2_5_2","SDK":"2.2.2-dev(c0eb301)"}}
10:59:48 RSL: stat/sonoff/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["hFE 2,4GHz",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","280500000100060000005A00000000000000","00000000"]}}
10:59:48 RSL: stat/sonoff/STATUS4 = {"StatusMEM":{"ProgramSize":562,"Free":440,"Heap":27704,"ProgramFlashSize":1024,"FlashSize":1024,"FlashChipId":"144051","FlashMode":3,"Features":["00000809","0FDAE396","041783A0","22B617CD","01003BC0"]}}
10:59:48 RSL: stat/sonoff/STATUS5 = {"StatusNET":{"Hostname":"schlafzimmer-rollladen","IPAddress":"192.168.0.53","Gateway":"192.168.0.254","Subnetmask":"255.255.255.0","DNSServer":"192.168.0.254","Mac":"60:01:94:9E:9C:18","Webserver":2,"WifiConfig":4}}
10:59:48 RSL: stat/sonoff/STATUS6 = {"StatusMQT":{"MqttHost":"","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_9E9C18","MqttUser":"DVES_USER","MqttCount":0,"MAX_PACKET_SIZE":1000,"KEEPALIVE":30}}
10:59:48 RSL: stat/sonoff/STATUS7 = {"StatusTIM":{"UTC":"Sun Jul 28 09:59:48 2019","Local":"Sun Jul 28 10:59:48 2019","StartDST":"Sun Mar 31 02:00:00 2019","EndDST":"Sun Oct 27 03:00:00 2019","Timezone":"+01:00","Sunrise":"05:18","Sunset":"20:35"}}
10:59:48 RSL: stat/sonoff/STATUS10 = {"StatusSNS":{"Time":"2019-07-28T10:59:48"}}
10:59:48 RSL: stat/sonoff/STATUS11 = {"StatusSTS":{"Time":"2019-07-28T10:59:48","Uptime":"0T00:14:12","Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":20,"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"hFE 2,4GHz","BSSId":"00:F7:6F:CE:D0:62","Channel":1,"RSSI":80,"LinkCount":1,"Downtime":"0T00:00:06","DeepSleep":0,"Heap":27664}}}
```
- [ ] Provide the output of console when you experience your issue if apply :
 (see above)

### TO REPRODUCE
Flash this fork's sonoff-minimal/firmware.bin on top of mainline Tasmota 6.5.0.8 on a Sonoff Dual R2

### EXPECTED BEHAVIOUR
The device should have reconnected to my AP. At the very least, it should have started wifi manager.

**Comment by stefanbode** [comment]
Indeed the two config files are not compatible to each other. This is hard to fix and a known limitation. Anyhow the expected behavior is that the newly flashed device starts with default configuration. Is you use a user_config_override you also can preinject the WiFi credentials. This allows you to access the device. That it did not start at all is bad an normally should not happen. The config is a binary and this means everything can happen.

**Comment by Crazor** [comment]
I updated two other devices, one is another Sonoff Dual R2 and the other a Shelly 2.5. This time, I used upstream Tasmota's sonoff-minimal.bin, which came up fine. I then uploaded the firmware I compiled and everything worked as expected. I read about it in another issue and decided to try it with devices that would be accessible, if inconvenient, and was lucky enough that they both upgraded flawlessly. I will stick to this way for now.
Maybe the -minimal build should always start WiFi Manager, regardless of what the config says.
